geode-cq/src/main/java/org/apache/geode/cache/query/cq/internal/CqServiceImpl.java

```diff
@@ -1347,15 +1347,14 @@ private void processEntryEvent(CacheEvent event, Profile localProfile, Profile[]
           boolean error = false;
           {
             try {
-              synchronized (cQuery) {
-                // Apply query on new value.
-                if (!cqUnfilteredEventsSet_newValue.isEmpty()) {
-                  executionStartTime = this.stats.startCqQueryExecution();
-
+              // Apply query on new value.
+              if (!cqUnfilteredEventsSet_newValue.isEmpty()) {
+                executionStartTime = this.stats.startCqQueryExecution();
+                synchronized (cQuery) {
                   b_cqResults_newValue =
                       evaluateQuery(cQuery, new Object[] {cqUnfilteredEventsSet_newValue});
-                  this.stats.endCqQueryExecution(executionStartTime);
                 }
+                this.stats.endCqQueryExecution(executionStartTime);
```

Comment:
Neat, any idea how much moving this out of the block improved overall? Just curious

Reply:
Hey @jhuynh1,

I don't have an answer primarily because I ran the performance tests using the whole `PR`, not just this particular change. My guess, though, is that moving these statistics calls outside of the lock didn't improve things that much (unless these calls are expensive, which doesn't seem to be the case).
Below is a quick comparison between the `baseline` (columns 0, 1, 2), `develop` without this PR included (columns 1, 2, 3) and `develop` with this PR included. Overall the performance gain seems to be around 1% (sill not the same as the baseline, though).

```
================================================================================
                                Comparison Report
                          Wed Mar 11 09:34:28 GMT 2020
================================================================================
TEST     STATSPEC              OP    #0    #1   #2    #3    #4    #5    #6    #7    #8 
scale082 putResponseTime       del  --- -1.01  --- -1.04 -1.03 -1.02   --- -1.01   --- 
         putsPerSecond         avg  --- -1.01  --- -1.04 -1.03 -1.03 -1.01 -1.02   --- 
         updateEventsPerSecond avg  --- -1.01  --- -1.04 -1.03 -1.02   --- -1.02   --- 
         updateLatency         del  --- -1.01  --- -1.03 -1.04 -1.03 -1.02 -1.02 -1.02 
                               AVG 0.00 -1.01 0.00 -1.03 -1.03 -1.02 -1.01 -1.02 -1.02 

================================================================================
                                 Java Source Key
0: support/9.9:6ffabd4dc6f36e9cc0b585a18da666f372bbb1b9 (2019-12-19 17:55:45 +0000)
1: support/9.9:6ffabd4dc6f36e9cc0b585a18da666f372bbb1b9 (2019-12-19 17:55:45 +0000)
2: support/9.9:6ffabd4dc6f36e9cc0b585a18da666f372bbb1b9 (2019-12-19 17:55:45 +0000)
3: develop:86defd40c6ba558b53753c51176e05f009bb32c3 (2020-03-10 21:56:49 +0000)
4: develop:86defd40c6ba558b53753c51176e05f009bb32c3 (2020-03-10 21:56:49 +0000)
5: develop:86defd40c6ba558b53753c51176e05f009bb32c3 (2020-03-10 21:56:49 +0000)
6: feature/GEODE-7863:a811505a23148187778df8300f71d9146efc5b5c (2020-03-10 08:49:42 +0000)
7: feature/GEODE-7863:a811505a23148187778df8300f71d9146efc5b5c (2020-03-10 08:49:42 +0000)
8: feature/GEODE-7863:a811505a23148187778df8300f71d9146efc5b5c (2020-03-10 08:49:42 +0000)

================================================================================
                                  Directory Key
0: baseline_run1
1: baseline_run2
2: baseline_run3
3: developWithoutCQFixes_run1
4: developWithoutCQFixes_run2
5: developWithoutCQFixes_run3
6: developWithCQFixes_run1
7: developWithCQFixes_run2
8: developWithCQFixes_run3
================================================================================
```